addons/mrp/models/mrp_production.py

```diff
@@ -855,17 +855,18 @@ def write(self, vals):
             del vals['move_byproduct_ids']
         if 'workorder_ids' in self:
             production_to_replan = self.filtered(lambda p: p.is_planned)
-        if 'move_raw_ids' in vals and self.state not in ['draft', 'cancel', 'done']:
-            # When adding a move raw, it should have the source location's `warehouse_id`.
-            # Before, it was handle by an onchange, now it's forced if not already in vals.
-            warehouse_id = self.location_src_id.warehouse_id.id
-            if vals.get('location_src_id'):
-                location_source = self.env['stock.location'].browse(vals.get('location_src_id'))
-                warehouse_id = location_source.warehouse_id.id
-            for move_vals in vals['move_raw_ids']:
-                command, _id, field_values = move_vals
-                if command == Command.CREATE and not field_values.get('warehouse_id', False):
-                    field_values['warehouse_id'] = warehouse_id
+        for move_str in ('move_raw_ids', 'move_finished_ids'):
+            if move_str in vals and self.state not in ['draft', 'cancel', 'done']:
+                # When adding a move raw/finished, it should have the source location's `warehouse_id`.
+                # Before, it was handle by an onchange, now it's forced if not already in vals.
+                warehouse_id = self.location_src_id.warehouse_id.id
+                if vals.get('location_src_id'):
+                    location_source = self.env['stock.location'].browse(vals.get('location_src_id'))
+                    warehouse_id = location_source.warehouse_id.id
+                for move_vals in vals[move_str]:
+                    command, _id, field_values = move_vals
+                    if command == Command.CREATE and not field_values.get('warehouse_id', False):
+                        field_values['warehouse_id'] = warehouse_id
```

Comment:
As you added one more indentation, I think that many indentations is now difficult to read. It should even reduce the diff as you're not modifying the indented lines.
Here's my suggestion:

```suggestion
            if move_str not in vals or self.state in ['draft', 'cancel', 'done']:
                continue
            # When adding a move raw/finished, it should have the source location's `warehouse_id`.
            # Before, it was handle by an onchange, now it's forced if not already in vals.
            warehouse_id = self.location_src_id.warehouse_id.id
            if vals.get('location_src_id'):
                location_source = self.env['stock.location'].browse(vals.get('location_src_id'))
                warehouse_id = location_source.warehouse_id.id
            for move_vals in vals[move_str]:
                command, _id, field_values = move_vals
                if command == Command.CREATE and not field_values.get('warehouse_id', False):
                    field_values['warehouse_id'] = warehouse_id
```